site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,380 @@
+
+### Test Library
+
+There are many JavaScript test libraries available, and everyone has their preference. This guide uses a simple library for demonstration so that you can generalize the tests to your favorite library. The tests below use [`tape`](https://github.com/ljharb/tape), which implements the [Test Anything Protocol (TAP)](https://en.wikipedia.org/wiki/Test_Anything_Protocol), a language-agnostic specification for running tests that's been in use since 1987. The tests also use [fetch-mock](https://www.wheresrhys.co.uk/fetch-mock/) to mock `fetch` calls from your lambda.
+
+Install the following in your test app terminal.
+
+```bash
+npm install --save-dev tape
+npm install --save-dev faucet # a little test-runner to give neat tape output
+npm install --save-dev fetch-mock
+npm install --save-dev jsonwebtoken # to decode the JWT
+npm install --save-dev uuid # to make a random user Id
+```
+
+{% include _callout-note.liquid content=" The `fetch()` method is available natively from Node LTS version 18. In earlier versions, `fetch` was provided by libraries, so many popular mocking libraries for `fetch` (such as [Nock](https://github.com/nock/nock)) won't work with modern Node in 2023." %}
+
+### Integration Test: Verify JWT Population
+
+The first of the two tests you're going to write is an integration test. It will verify that your updated lambda is populating the JWT with a "Goodbye World" message when you log in programmatically.
+
+#### Create A User
+
+Before you can write any tests, you need a test user profile to log in with. The test app `package.json` includes a reference to the `@fusionauth/typescript-client` discussed earlier, so you can use that to create a test user programmatically.
+
+Make a new file called `test.js` in the app and paste the following code into it.
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_1.js' %}
+```
+
+The code above has three sections:
+
+- A declaration of constant variables that match the `kickstart.json` and `.env` files.
+- The `createRandomUser` function, which creates a User request object with `const request` and then sends it to `fusion.register()`. Details on this can be found in the [TypeScript client library interface](https://github.com/FusionAuth/fusionauth-typescript-client/blob/master/src/FusionAuthClient.ts).
+- The `deleteUser` function, which you can use in the tests to delete the user just created.
+
+Run the code to test user creation with the following command.
+
+```bash
+node test.js
+```
+
+In FusionAuth, click on <span>Users</span>{:.breadcrumb} to check that a new user called `lambdatestuser` has been created. You can delete the `createRandomUser(uuidv4());` line in `test.js` as each test will use a new temporary user. This will allow you to add multiple lambda tests while avoiding potential conflicts between test users and permissions.
```

Comment:
why did we call `createRandomUser` here just to delete it? Is this proving that the typescript library is configured correctly or for some other reason?